src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -243,20 +259,73 @@ def rook_api_post(self, path: str, **kwargs: Any) -> Any:
     def get_discovered_devices(self, nodenames: Optional[List[str]] = None) -> Dict[str, dict]:
         def predicate(item: client.V1ConfigMapList) -> bool:
             if nodenames is not None:
-                return item.metadata.labels['rook.io/node'] in nodenames
+                return item['spec']['nodeName'] in nodenames
             else:
                 return True
+        matching_sc = [i for i in self.storage_classes.items if self.storage_class == i.metadata.name]
+        if len(matching_sc) == 0:
+            log.exception("no storage class exists matching name provided in ceph config at mgr/rook/storage_class")
+            raise Exception('No storage class exists matching name provided in ceph config at mgr/rook/storage_class')
+        if len(matching_sc) > 1:
+            log.exception("too many storage classes matching name provided in ceph config at mgr/rook/storage_class")
+            raise Exception('Too many storage classes matching name provided in ceph config at mgr/rook/storage_class')
```

Comment:
I suggest to inform the user of the storage class used in the rook module. Not redirect the user to see what is this value when we can inform about it very easily . As example,  Probably a message like:
`f"No storage class exists matching configured Rook orchestrator storage class which currently is <{self.storage_class}>. This storage class can be set in ceph config (mgr/rook/storage_class)"`
is more useful